# Understanding Literary Data

## ScriptoNet Digital Humanities Project

This notebook explores how literary texts can be transformed into structured data for computational analysis.

The purpose is to understand:

- How novels are represented as digital text
- How text can be processed
- How passages can become research units
- How literary data can later be used for NLP models

This is the first step before developing Artificial Intelligence models.

## Research Question

How can a literary narrative be converted into structured data while preserving meaningful units for computational analysis?

A computer does not directly understand:

- characters
- emotions
- healing
- narrative meaning

It processes:

- characters
- words
- sentences
- patterns

Therefore, the first challenge is transforming literature into a suitable computational format.

In [2]:
import pandas as pd
import re
import matplotlib.pyplot as plt

## Dataset Selection

For initial experiments, we use:

Mary Shelley's *Frankenstein*

Reasons:

- Public domain text
- Easily reproducible
- Contains narrative structures relevant to ScriptoNet:
  - letters
  - memories
  - personal reflections
  - emotional transformation

The purpose is not to study Frankenstein alone, but to develop the computational pipeline.

## Loading Literary Text Data

In this section, we load a public domain literary text and prepare it for computational analysis.

The purpose is to understand how a literary work can be transformed into a structured dataset before applying Natural Language Processing methods.

In [3]:
import requests

url = "https://www.gutenberg.org/files/84/84-0.txt"

response = requests.get(url)

text = response.text

print(text[:1000])

*** START OF THE PROJECT GUTENBERG EBOOK 84 ***

Frankenstein;

or, the Modern Prometheus

by Mary Wollstonecraft (Godwin) Shelley


 CONTENTS

 Letter 1
 Letter 2
 Letter 3
 Letter 4
 Chapter 1
 Chapter 2
 Chapter 3
 Chapter 4
 Chapter 5
 Chapter 6
 Chapter 7
 Chapter 8
 Chapter 9
 Chapter 10
 Chapter 11
 Chapter 12
 Chapter 13
 Chapter 14
 Chapter 15
 Chapter 16
 Chapter 17
 Chapter 18
 Chapter 19
 Chapter 20
 Chapter 21
 Chapter 22
 Chapter 23
 Chapter 24




Letter 1

_To Mrs. Saville, England._


St. Petersburgh, Dec. 11th, 17—.


You will rejoice to hear that no disaster has accompanied the
commencement of an enterprise which you have regarded with such evil
forebodings. I arrived here yesterday, and my first task is to assure
my dear sister of my welfare and increasing confidence in the success
of my undertaking.

I am already far north of London, and as I walk in the streets of
Petersburgh, I feel a cold northern breeze play upon my cheeks, which
braces my nerves and fills me w

## Basic Text Exploration

Before applying Natural Language Processing methods, we first examine the basic properties of the literary text.

This helps us understand the size, structure, and characteristics of the dataset.

In [4]:
# Number of characters
characters = len(text)

# Split text into words
words = text.split()

# Number of words
word_count = len(words)

print("Total characters:", characters)
print("Total words:", word_count)

Total characters: 419434
Total words: 75060


## Observation

The literary text has been successfully converted into digital form.

The first exploration shows:

- Total characters: 419,434
- Total words: 75,060

At this stage, the computer only understands the text as a sequence of characters and words.

However, literary meaning such as trauma, healing, memory, and emotional transformation is not directly available from raw text.

The next challenge is to transform this continuous text into meaningful computational units while preserving enough context for literary analysis.

## Passage Segmentation

Machine learning models cannot directly analyse an entire novel as a single unit.

Therefore, the literary text is divided into smaller passages.

Each passage becomes a potential research sample that can later be annotated and used for NLP experiments.

The goal is to preserve enough context for literary interpretation while creating manageable computational units.

In [5]:
# Split text into paragraphs

paragraphs = text.split("\n\n")

# Remove empty paragraphs
paragraphs = [
    p.strip()
    for p in paragraphs
    if p.strip()
]

print("Total paragraphs:", len(paragraphs))

# Show first paragraph
print(paragraphs[0])

Total paragraphs: 797
*** START OF THE PROJECT GUTENBERG EBOOK 84 ***



## Observation: Text Cleaning Requirement

The raw text contains non-literary information such as Project Gutenberg metadata.

Example:

"*** START OF THE PROJECT GUTENBERG EBOOK 84 ***"

This information is not part of the literary narrative and should be removed before computational analysis.

This demonstrates an important challenge in Digital Humanities:

Computers process all available text, but researchers must identify which parts contain meaningful literary information.

## Removing Non-Literary Metadata

Before analysing literary patterns, we remove unnecessary text markers from the digital source.

The goal is to keep only the actual narrative content.

In [6]:
# Remove Gutenberg header and footer information

start_marker = "*** START OF THE PROJECT GUTENBERG EBOOK 84 ***"
end_marker = "*** END OF THE PROJECT GUTENBERG EBOOK 84 ***"

clean_text = text.split(start_marker)[1]
clean_text = clean_text.split(end_marker)[0]

clean_text = clean_text.strip()

print(clean_text[:1000])

Frankenstein;

or, the Modern Prometheus

by Mary Wollstonecraft (Godwin) Shelley


 CONTENTS

 Letter 1
 Letter 2
 Letter 3
 Letter 4
 Chapter 1
 Chapter 2
 Chapter 3
 Chapter 4
 Chapter 5
 Chapter 6
 Chapter 7
 Chapter 8
 Chapter 9
 Chapter 10
 Chapter 11
 Chapter 12
 Chapter 13
 Chapter 14
 Chapter 15
 Chapter 16
 Chapter 17
 Chapter 18
 Chapter 19
 Chapter 20
 Chapter 21
 Chapter 22
 Chapter 23
 Chapter 24




Letter 1

_To Mrs. Saville, England._


St. Petersburgh, Dec. 11th, 17—.


You will rejoice to hear that no disaster has accompanied the
commencement of an enterprise which you have regarded with such evil
forebodings. I arrived here yesterday, and my first task is to assure
my dear sister of my welfare and increasing confidence in the success
of my undertaking.

I am already far north of London, and as I walk in the streets of
Petersburgh, I feel a cold northern breeze play upon my cheeks, which
braces my nerves and fills me with delight. Do you understand this
feeling? This

## Observation: Literary Structure Matters

After removing metadata, the text reveals its original narrative structure.

Frankenstein begins with a series of letters before moving into the main narrative.

This demonstrates an important challenge for computational literary analysis:

The same textual form can have different meanings depending on context.

For example:

A letter may represent:
- simple communication
- narrative framing
- emotional reflection
- confession or therapeutic processing

Therefore, AI models must learn contextual meaning rather than only detecting keywords.

## Creating Computational Passages

Machine learning models require structured examples.

A complete novel is too large and complex to use directly.

Therefore, the text is divided into smaller passages.

Each passage becomes an individual unit for future annotation and AI analysis.

The goal is to maintain enough literary context while creating manageable data samples.

In [7]:
# Create passages from cleaned paragraphs

paragraphs = [
    p.strip()
    for p in clean_text.split("\n\n")
    if p.strip()
]

print("Number of cleaned paragraphs:", len(paragraphs))


# Combine paragraphs into larger passages

passages = []

current_passage = ""

for paragraph in paragraphs:
    current_passage += " " + paragraph

    # approximately create passages of 200-300 words
    if len(current_passage.split()) >= 250:
        passages.append(current_passage.strip())
        current_passage = ""

# Add remaining text
if current_passage:
    passages.append(current_passage.strip())


print("Total passages created:", len(passages))

Number of cleaned paragraphs: 795
Total passages created: 229


## Observation: From Literature to Dataset

The literary text has been transformed into structured computational units.

The original novel contained continuous narrative flow. For computational analysis, it was divided into 229 passages.

Each passage can now become an independent research sample.

However, segmentation alone does not create understanding.

A computer can identify passages, but it does not know whether a passage represents:

- trauma
- healing
- reflection
- confession
- emotional transformation

Human annotation is required to provide meaningful interpretation.

In [8]:
import pandas as pd

# Create dataframe

df = pd.DataFrame({
    "passage_id": range(1, len(passages)+1),
    "text": passages
})

df.head()

,passage_id,text
0,1,"Frankenstein; or, the Modern Prometheus by Mar..."
1,2,These reflections have dispelled the agitation...
2,3,Six years have passed since I resolved on my p...
3,4,This is the most favourable period for travell...
4,5,But I have one want which I have never yet bee...


In [9]:
df.to_csv(
    "frankenstein_passages.csv",
    index=False
)

print("Dataset saved successfully")

Dataset saved successfully


# First Dataset Creation Result

The first ScriptoNet experimental dataset was created from Mary Shelley's Frankenstein.

Dataset statistics:

- Total words in raw text: 75,060
- Total characters: 419,434
- Cleaned paragraphs: 795
- Generated passages: 229

Each passage represents a potential unit for future human annotation and machine learning experiments.

At this stage, the dataset contains only literary text. It does not yet contain therapeutic labels.

The next research challenge is creating meaningful annotations that connect literary interpretation with computational learning.